src/rust/engine/src/nodes.rs

```diff
@@ -590,7 +571,10 @@ impl SelectTransitive {
           selector: selectors::Select { product: self.selector.product, variant_key: None },
           subject: subject_key,
           variants: self.variants.clone(),
-          entries: self.product_entries.clone()
+          // NB: Filtering by the matching subject type, since we now have a concrete one.
```

Comment:
Can you add a paired comment somewhere else to show the contrast between "now" and "then"? It's not clear from this comment where it is that we don't know the subject type.